Source/CarthageKit/Project.swift

```diff
@@ -639,33 +653,87 @@ public final class Project {
 			.zipWith(submodulesSignal)
 			.flatMap(.Merge) { dependencies, submodulesByPath -> SignalProducer<(), CarthageError> in
 				return SignalProducer<Dependency<PinnedVersion>, CarthageError>(values: dependencies)
-					.flatMap(.Merge) { dependency -> SignalProducer<(), CarthageError> in
+					.flatMap(.Concat) { dependency -> SignalProducer<(), CarthageError> in
 						let project = dependency.project
-						let revision = dependency.version.commitish
 
 						let submoduleFound = submodulesByPath[project.relativePath] != nil
-						let checkoutOrCloneProject = self.checkoutOrCloneProject(project, atRevision: revision, submodulesByPath: submodulesByPath)
+						let checkoutOrCloneDependency = self.checkoutOrCloneDependency(dependency, submodulesByPath: submodulesByPath)
 
 						// Disable binary downloads for the dependency if that
 						// is already checked out as a submodule.
 						if submoduleFound {
-							return checkoutOrCloneProject
+							return checkoutOrCloneDependency
 						}
 
-						return self.installBinariesForProject(project, atRevision: revision)
+						return self.installBinariesForProject(project, atRevision: dependency.version.commitish)
 							.flatMap(.Merge) { installed -> SignalProducer<(), CarthageError> in
 								if installed {
 									return .empty
 								} else {
-									return checkoutOrCloneProject
+									return checkoutOrCloneDependency
 								}
 							}
 					}
 			}
 			.then(.empty)
 	}
 
-	/// Attempts to build each Carthage dependency that has been checked out, 
+	/// Creates symlink between the dependency checkouts and the root checkouts
+	private func symlinkCheckoutPathsForDependencyProject(dependency: ProjectIdentifier, subDependencies: Set<ProjectIdentifier>, rootDirectoryURL: NSURL) -> SignalProducer<(), CarthageError> {
+		let rootCheckoutsURL = rootDirectoryURL.URLByAppendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).URLByResolvingSymlinksInPath!
+		let rawDependencyURL = rootDirectoryURL.URLByAppendingPathComponent(dependency.relativePath, isDirectory: true)
+		let dependencyURL = rawDependencyURL.URLByResolvingSymlinksInPath!
+		let dependencyCheckoutsURL = dependencyURL.URLByAppendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).URLByResolvingSymlinksInPath!
+		let subDependencyNames = subDependencies.map { $0.name }
+		let fileManager = NSFileManager.defaultManager()
+
+		let symlinksProducer =  SignalProducer(values: Array(subDependencyNames))
```

Comment:
Ah, I had it in my head it was a set because the original subDependencies was. Done.